default/scripting/buildings/GAS_GIANT_GEN.focs.txt

```diff
@@ -16,29 +16,37 @@ BuildingType
             scope = And [
                 Planet
                 InSystem id = Source.SystemID
-                Planet type = GasGiant
                 Focus type = "FOCUS_INDUSTRY"
                 NOT Population high = 0
                 OwnedBy empire = Source.Owner
             ]
+            activation = ContainedBy And [
+                Planet
+                Population high = 0
+            ]
             stackinggroup = "GAS_GIANT_GEN_STACK"
             priority = [[VERY_LATE_PRIORITY]]
-            effects = SetTargetIndustry value = Value + 5
+            effects = SetTargetIndustry value = Value + 10
 
         EffectsGroup
             scope = And [
                 Planet
                 InSystem id = Source.SystemID
-                Not Planet type = GasGiant
                 Focus type = "FOCUS_INDUSTRY"
                 NOT Population high = 0
                 OwnedBy empire = Source.Owner
             ]
+            activation = ContainedBy And [
+                Planet
+                Not Population high = 0
+            ]
             stackinggroup = "GAS_GIANT_GEN_STACK"
-            priority = [[VERY_LATE_PRIORITY]]
-            effects = SetTargetIndustry value = Value + 10
+            // if there is an unpopulated gas giant generator planet then that bonus applies
+            // VERY_LATE_PRIORITY__PLUS_ONE
+            priority = 121
```

Comment:
Making the priority for the populated case be different than the priority for the non-populated case seems to me to invite trouble